modules/carto/src/index.ts

```diff
@@ -29,6 +32,22 @@ export {
   VectorTileLayer
 };
 
+// Internal Layers
+export {default as _QuadbinLayer} from './layers/quadbin-layer';
+export {default as _RasterLayer} from './layers/raster-layer';
+export {default as _SpatialIndexTileLayer} from './layers/spatial-index-tile-layer';
+
+// Types
+export type {ClusterTileLayerProps} from './layers/cluster-tile-layer';
+export type {H3TileLayerProps} from './layers/h3-tile-layer';
+export type {HeatmapTileLayerProps} from './layers/heatmap-tile-layer';
+export type {PointLabelLayerProps} from './layers/point-label-layer';
+export type {QuadbinTileLayerProps} from './layers/quadbin-tile-layer';
+export type {RasterTileLayerProps} from './layers/raster-tile-layer';
+export type {SpatialIndexTileLayerProps} from './layers/spatial-index-tile-layer';
+export type {VectorTileLayerProps} from './layers/vector-tile-layer';
```

Comment:
Optional - do we also want to export QuadbinLayerProps and RasterLayerProps? And, does it matter if the props type for an 'internal' layer is not prefixed, but the layer constructor is? (`_SpatialIndexTileLayer` vs `SpatialIndexTileLayerProps` here).

No change needed, but if it's clearer we could also organize these in pairs to make sure we include props for each layer:

```javascript
import {default as ClusterTileLayer} from './layers/cluster-tile-layer';
export type {ClusterTileLayerProps} from './layers/cluster-tile-layer';

import {default as H3TileLayer} from './layers/h3-tile-layer';
export type {H3TileLayerProps} from './layers/h3-tile-layer';

...
```

Reply:
Good catch on the types, I meant to include those. Regarding the structure, I went for the same pattern adopted in `modules/layers/index.ts`.

On the `_Type` pattern: many internal types can be surfaced by the compiler even when they are not exported, and it would look ugly and inconsistent to have them marked with underscore. On the other hand, classes which are designated as internal and could change without warning in the future are better marked with an underscore to warn the user.